components/script/script_thread.rs

```diff
@@ -2918,60 +2902,32 @@ impl ScriptThread {
 
     /// Handles a request to exit a pipeline and shut down layout.
     fn handle_exit_pipeline_msg(&self, id: PipelineId, discard_bc: DiscardBrowsingContext) {
-        debug!("Exiting pipeline {}.", id);
+        debug!("{id}: Starting pipeling exit.");
 
         self.closed_pipelines.borrow_mut().insert(id);
 
-        // Check if the exit message is for an in progress load.
-        let idx = self
-            .incomplete_loads
-            .borrow()
-            .iter()
-            .position(|load| load.pipeline_id == id);
-
-        let document = self.documents.borrow_mut().remove(id);
-
         // Abort the parser, if any,
         // to prevent any further incoming networking messages from being handled.
-        if let Some(document) = document.as_ref() {
+        let document = self.documents.borrow_mut().remove(id);
+        if let Some(document) = document {
             if let Some(parser) = document.get_current_parser() {
                 parser.abort();
             }
-        }
-
-        // We should never have a pipeline that's still an incomplete load,
-        // but also has a Document.
-        debug_assert!(idx.is_none() || document.is_none());
```

Comment:
It seems like this assertion could be retained as it could be helpful in checking the invariant, even though `idx` is no longer needed to get the layout for incomplete loads.

Reply:
I've re-added this assertion.